src/audio/src/src.c

```diff
@@ -545,6 +583,409 @@ static void src_set_sink_params(struct comp_dev *dev, struct comp_buffer __spars
 	sinkb->buffer_fmt = cd->ipc_config.base.audio_fmt.interleaving_style;
 }
 
+static inline void src_update_buffer_position(struct input_stream_buffer *input_buffers,
+					      struct output_stream_buffer *output_buffers,
+					      int *n_read, int *n_written)
+{
+	struct audio_stream __sparse_cache *source = input_buffers->data;
+	struct audio_stream __sparse_cache *sink = output_buffers->data;
+
+	input_buffers->consumed += audio_stream_frame_bytes(source) * (*n_read);
+	output_buffers->size += audio_stream_frame_bytes(sink) * (*n_written);
+}
+
+static int src_verify_params(struct comp_dev *dev,
+			     struct sof_ipc_stream_params *params)
+{
```

Comment:
you're still duplicating code. What I meant was:
```
@@ -624,10 +624,10 @@ static void src_free(struct comp_dev *dev)
 	rfree(dev);
 }
 
-static int src_verify_params(struct comp_dev *dev,
-			     struct sof_ipc_stream_params *params)
+static int src_verify_params_intern(struct comp_dev *dev,
+				    struct sof_ipc_stream_params *params,
+				    struct comp_data *cd)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
 	int ret;
 
 	comp_dbg(dev, "src_verify_params()");
@@ -661,6 +661,14 @@ static int src_verify_params(struct comp_dev *dev,
 	return ret;
 }
 
+static int src_verify_params(struct comp_dev *dev,
+			     struct sof_ipc_stream_params *params)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	return src_verify_params_intern(dev, params, cd);
+}
+
 /* set component audio stream parameters */
 static int src_params(struct comp_dev *dev,
 		      struct sof_ipc_stream_params *params)
```
and then for the module adapter version:
```
+static int src_verify_params(struct comp_dev *dev,
+			     struct sof_ipc_stream_params *params)
+{
+	struct processing_module *mod = comp_get_drvdata(dev);
+	struct comp_data *cd = module_get_private_data(mod);
+
+	return src_verify_params_intern(dev, params, cd);
+}
```
This should let you make the diff quite small and also avoid adding duplicate code to the file.

Reply:
thanks for your advice. For the convenience of review, I split the previous commit into several.